fairseq/training.py

```diff
@@ -0,0 +1,434 @@
+                        local_unzipped_dir.append(os.path.join(os.path.dirname(cached_audio_zip_dir), "audio"))
+                        sp.check_call(
+                            [
+                                "unzip",
+                                "-q",
+                                "-n",
+                                cached_audio_zip_dir,
+                                "-d",
+                                local_unzipped_dir[-1],
+                            ]
+                        )
+                    except sp.CalledProcessError:
+                        print(
+                            f"Cache manager: Error occurred for caching and unzipping audio data in {local_unzipped_dir[-1]}"
+                        )
+                        raise
+                common_audio_dir = os.path.commonpath(local_unzipped_dir)
+                for name in ["train.tsv", "valid.tsv"]:
+                    with open(os.path.join(manifest_path, name), "r") as manifest_file:
+                        manifest_lines = manifest_file.read().splitlines()
+                    for i in range(1, len(manifest_lines)):
+                        to_check = (
+                            os.path.join(common_audio_dir, manifest_lines[i].split()[0])
+                        )
+                        assert os.path.exists(
+                            to_check
+                        ), f"Manifest file {to_check} not found in unzipped directory"
+                    manifest_lines[0] = common_audio_dir
+                    with open(
+                        os.path.join(self.out_cached_audio_manifest.get_path(), name), "w"
+                    ) as cached_audio_manifest_file:
+                        cached_audio_manifest_file.write("\n".join(manifest_lines))
+
+        my_env = os.environ
+        if self.fairseq_root is not None:
+            my_env["PYTHONPATH"] = self.fairseq_root
+        sp.check_call(self._get_run_cmd(), env=my_env)
+
+    def plot(self):
+        directory = "./outputs"
+        train_score, train_error = {}, {}
+        valid_score, valid_error = {}, {}
+        learning_rates = {}
+
+        for cur in os.walk(directory):
+            dir_path = cur[0]
+            files = cur[2]
+            if "hydra_train.log" in files:
+                with open(f"{dir_path}/hydra_train.log", "r") as f:
+                    lines = f.readlines()
+                    for i in range(len(lines) - 1):
+                        line = lines[i]
+                        if 'begin validation on "valid" subset' in line:
+                            split = re.sub('[{,":]', "", lines[i + 1]).split(" ")
+                            #  Exception handling for cases in which accuracy and loss 
+                            #  are not logged in next line
+                            try:
+                                epoch = int(split[split.index("epoch") + 1])
+                                score = float(split[split.index("valid_loss") + 1])
```

Comment:
Similarly, it can be "dev_loss" based on the name of the validation set. ie if validation set is "valid"then it is "valid_loss" but if the name is "dev" then it is "dev_loss"

Reply:
good point. If the line contains a printed dict as @nishthajain1611 indicated below, then you could maybe try to parse it like
```
epoch_dict = eval(line[line.index("{"):])
epoch = epoch_dict[epoch]
losses = {k:v for k,v in epoch_dict.items() if k.endswith("_loss")}
erros = {k:v for k,v in epoch_dict.items() if k.endswith("_error")}
```